ome/ome-files-py
# Switch to pybind11

Reimplements the extension module with pybind11. This brings the C++ code to 127 lines, down from about 600.

The idea was to replicate the exact functionality of the extension module while leaving all Python code unchanged, including tests. The only deviation from this is that I've removed the custom `ome_files.Error` exception in favor of pybind11's automatic exception translation. This makes the C++ code much simpler and there is also an improvement on the Python side where the underlying exception is more significant than a generic error, see for instance the translation to `IndexError` in `get_index`.

Another minor improvement is that now we're supporting both signatures for `is_interleaved`.

Finally, I've exposed the upstream version to address #21:

```
>>> import ome_files
>>> ome_files.upstream_version_info
Version(major=0L, minor=3L, patch=2L, extra=u'-DEV-df1a632')
>>> ome_files.upstream_version
'0.3.2-DEV-df1a632'
```

**Testing:**
* check that the Travis build stays green and that there are no major crimes in the new C++ code
* check that that the superbuild (together with https://github.com/ome/ome-cmake-superbuild/pull/155) does not break
